app/websocket/app/main/model_manager.py

```diff
@@ -163,10 +202,8 @@ def save_model(serialized_model: bytes, model_id: str):
     except (SQLAlchemyError, IntegrityError) as e:
         if type(e) is IntegrityError:
             # The model is already present within the db.
-            # But missing from cache. Fetch the model and save to cache.
-            db_model = get_model_with_id(model_id)
```

Comment:
needed to handle the case where:
- Db saving fails and exception is raised cause model already exists.
- Better to cache such a model... as it might be used next.
- kind of pre emptive catching + saves a db call.

Reply:
* Db saving fails and exception is raised cause model already exists.

If this is the case then an IntegrityError would occur before.

* Better to cache such a model... as it might be used next.

This is already done at get_model_with_id that's why I remove this code.

* kind of pre emptive catching + saves a db call.

What do you mean?